task/host-sp-comms/src/bsp/gimlet_bcde.rs

```diff
@@ -386,31 +403,36 @@ impl ServerImpl {
             }
 
             52..=57 => {
-                let i = index as usize - 52;
-                // XXX this assumes that designator order is matched in the TOML
-                // file and in our list of names below!
-                let dev = i2c_config::devices::tmp117(I2C.get_task_id())[i];
-                let name = match i {
-                    0 => b"J194/U1",
-                    1 => b"J195/U1",
-                    2 => b"J196/U1",
-                    3 => b"J197/U1",
-                    4 => b"J198/U1",
-                    5 => b"J199/U1",
+                let (connector_name, f, sensors) = match index - 52 {
+                    0 => by_refdes!(J194, tmp117),
+                    1 => by_refdes!(J195, tmp117),
+                    2 => by_refdes!(J196, tmp117),
+                    3 => by_refdes!(J197, tmp117),
+                    4 => by_refdes!(J198, tmp117),
+                    5 => by_refdes!(J199, tmp117),
                     _ => unreachable!(),
                 };
+                let dev = f(I2C.get_task_id());
+
+                // Convert the name from Jxxx (in the TOML file) -> Jxxx/U1
+                let mut name = *b"Jxxx/U1";
+                name[..connector_name.len()].copy_from_slice(connector_name);
```

Comment:
If we accidentally had a `connector_name` with length != 4, this would silently use that length, and we'd end up with `J12x/U1` (if short) or `J12345U1` (if long, or a panic if it was very long), right? Would it be slightly safer to do something like

```rust
                // All connector names should have length 4; using
                // a literal 4 here asserts that via the bounds check.
                name[..4].copy_from_slice(connector_name);
```

Reply:
I found a sneaky way to ensure that we actually have a fixed-size byte array here; done in fe2c235f7aefd8f8b7875edbb7c09f230948bfc7

Reply:
Neat!